rosbag2_transport/test/rosbag2_transport/test_record_topic_types.cpp

```diff
@@ -0,0 +1,85 @@
+// Copyright 2024, Open Source Robotics Corporation.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <gmock/gmock.h>
+
+#include <chrono>
+#include <memory>
+#include <regex>
+#include <string>
+#include <utility>
+
+#include "rclcpp/rclcpp.hpp"
+
+#include "rosbag2_test_common/publication_manager.hpp"
+#include "rosbag2_test_common/wait_for.hpp"
+#include "rosbag2_test_common/client_manager.hpp"
+
+#include "rosbag2_transport/recorder.hpp"
+
+#include "test_msgs/msg/arrays.hpp"
+#include "test_msgs/msg/basic_types.hpp"
+#include "test_msgs/message_fixtures.hpp"
+#include "test_msgs/srv/basic_types.hpp"
+
+#include "record_integration_fixture.hpp"
```

Comment:
Clean up in header includes. Removed unused and sorted out:
```suggestion
#include "record_integration_fixture.hpp"
#include "rosbag2_test_common/publication_manager.hpp"
#include "rosbag2_test_common/wait_for.hpp"
#include "rosbag2_transport/recorder.hpp"
#include "test_msgs/message_fixtures.hpp"
```